### PR TITLE
add unit test for getArgsAsStringInputFromForm.

### PR DESCRIPTION
**Pull request by iossocket**
# Add unit test for getArgsAsStringInputFromForm.

## Types of change

- [ ] Feature
- [ ] Bug
- [x] Enhancement

### CODE REVIEW

File: packages/nextjs/utils/scaffold-stark/types.ts [hunk 1]
```diff
@@ -17,10 +17,10 @@ import type {
 import { AbiEnum, AbiStruct } from "./contract";
 
 export const isCairoInt = (type: string): type is CairoInt =>
-  /^core::integer::(u|i)(8|16|32)$/.test(type);
+  /core::integer::(u|i)(8|16|32)$/.test(type);
```
Comment (metalboyrick) [inline packages/nextjs/utils/scaffold-stark/types.ts:20]: what is the difference if we remove the `^`

File: packages/nextjs/app/debug/_components/contract/__test__/utilsContract.test.tsx [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+              "0": 1,
+              "1": 2
+            },
+          }
+        }
+      }
+    ])
+  });
+
+  it("should parse multi input correctly", () => {
```
Comment (metalboyrick) [inline packages/nextjs/app/debug/_components/contract/__test__/utilsContract.test.tsx:84]: This is called an option / enum

File: packages/nextjs/app/debug/_components/contract/__test__/utilsContract.test.tsx [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+    expect(getArgsAsStringInputFromForm({ "echo_u8_input_core::bool": '0x01' }, false, true)).toEqual([true]);
+    expect(getArgsAsStringInputFromForm({ "echo_u8_input_core::bool": '0x001' }, false, true)).toEqual([true]);
+
+    expect(getArgsAsStringInputFromForm({ "echo_u8_input_core::bool": 'false' }, false, true)).toEqual([false]);
+    expect(getArgsAsStringInputFromForm({ "echo_u8_input_core::bool": '0' }, false, true)).toEqual([false]);
+    expect(getArgsAsStringInputFromForm({ "echo_u8_input_core::bool": '0x0' }, false, true)).toEqual([false]);
+    expect(getArgsAsStringInputFromForm({ "echo_u8_input_core::bool": '0x00' }, false, true)).toEqual([false]);
+    expect(getArgsAsStringInputFromForm({ "echo_u8_input_core::bool": '0x000' }, false, true)).toEqual([false]);
+  });
+
```
Comment (metalboyrick) [inline packages/nextjs/app/debug/_components/contract/__test__/utilsContract.test.tsx:34]: would request one for tuples as well , other than that we're good

**Review by metalboyrick — COMMENTED** [review]
Overall awesome job ser ! really appreciate the unit tests as we really need these for the parsers. Fo your context we used to implement our own parsers, but now we are shifting to `starknetjs` ones if possible. Left minor comments and some curious questions

File: packages/nextjs/app/debug/_components/contract/__test__/utilsContract.test.tsx (hunk 3 above)
Comment (iossocket) [reply packages/nextjs/app/debug/_components/contract/__test__/utilsContract.test.tsx:34]: sure, will add it

File: packages/nextjs/utils/scaffold-stark/types.ts (hunk 1 above)
Comment (iossocket) [reply packages/nextjs/utils/scaffold-stark/types.ts:20]: the type string passed in here may have some prefix(e.g. function name prefix `echo_int_intput_u16_core::integer::u16`),  removing`^` will only check the end part

**Review by metalboyrick — APPROVED** [review]
lgtm

**Pull request merged**